exporter/opentelemetry-exporter-prometheus/src/opentelemetry/exporter/prometheus/__init__.py

```diff
@@ -159,96 +162,153 @@ def collect(self) -> None:
 
         metric_family_id_metric_family = {}
 
-        while self._metrics_to_export:
-            for export_record in self._metrics_to_export.popleft():
-                self._translate_to_prometheus(
-                    export_record, metric_family_id_metric_family
-                )
+        while self._metrics_datas:
+            self._translate_to_prometheus(
+                self._metrics_datas.popleft(), metric_family_id_metric_family
+            )
 
             if metric_family_id_metric_family:
                 for metric_family in metric_family_id_metric_family.values():
                     yield metric_family
 
+    # pylint: disable=too-many-locals,too-many-branches
     def _translate_to_prometheus(
         self,
-        metric: Metric,
+        metrics_data: MetricsData,
         metric_family_id_metric_family: Dict[str, PrometheusMetric],
     ):
-        label_values = []
-        label_keys = []
-        for key, value in metric.attributes.items():
-            label_keys.append(self._sanitize(key))
-            label_values.append(self._check_value(value))
-
-        metric_name = ""
-        if self._prefix != "":
-            metric_name = self._prefix + "_"
-        metric_name += self._sanitize(metric.name)
-
-        description = metric.description or ""
-
-        metric_family_id = "|".join(
-            [metric_name, description, "%".join(label_keys), metric.unit]
-        )
-
-        if isinstance(metric.point, Sum):
-
-            metric_family_id = "|".join(
-                [metric_family_id, CounterMetricFamily.__name__]
-            )
-
-            if metric_family_id not in metric_family_id_metric_family.keys():
-                metric_family_id_metric_family[
-                    metric_family_id
-                ] = CounterMetricFamily(
-                    name=metric_name,
-                    documentation=description,
-                    labels=label_keys,
-                    unit=metric.unit,
+        metrics = []
+
+        for resource_metrics in metrics_data.resource_metrics:
```

Comment:
What happens to resource attributes? and instrumentation scope name and version?

Reply:
This has not changed, they were not being used in the translation before as well.

Reply:
I think the prev implementation had the assumption that the flat struct included all the attributes including the resource ones'. Would be great to open a separate issue to address this.

Reply:
I mean, all the resource information, including its attributes is still there for the prometheus exporter to access, since a `MetricsData` object is passed to it which includes the complete resource.